src/System.Net.Primitives/tests/FunctionalTests/NetworkCredentialTest.netstandard1.7.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Security;
+using Xunit;
+
+namespace System.Net.Primitives.Functional.Tests
+{
+    public static partial class NetworkCredentialTest
+    {
+        public static IEnumerable<object[]> SecurePassword_TestData()
+        {
+            yield return new object[] { "password", AsSecureString("password") };
+            yield return new object[] { "SecurePassword", AsSecureString("SecurePassword") };
+            yield return new object[] { "", AsSecureString("") };
+            yield return new object[] { null, AsSecureString(null) };
+        }
+
+        [Fact]
+        public static void Ctor_SecureString_Test()
+        {
+            string expectedUser = "UserName";
+
+            using (SecureString expectedSecurePassword = AsSecureString("password"))
+            {
+                NetworkCredential nc = new NetworkCredential(expectedUser, expectedSecurePassword);
+                Assert.Equal(expectedUser, nc.UserName);
+                Assert.True(expectedSecurePassword.CompareSecureString(nc.SecurePassword));
+            }
+        }
+
+        [Fact]
+        public static void Ctor_SecureStringDomain_Test()
+        {
+            string expectedUser = "UserName";
+            string expectedDomain = "thisDomain";
+
+            using (SecureString expectedSecurePassword = AsSecureString("password"))
+            {
+                NetworkCredential nc = new NetworkCredential(expectedUser, expectedSecurePassword, expectedDomain);
+                Assert.Equal(expectedUser, nc.UserName);
+                Assert.Equal(expectedDomain, nc.Domain);
+                Assert.True(expectedSecurePassword.CompareSecureString(nc.SecurePassword));
+            }
+        }
```

Comment:
Related to my other comments, we need to add tests for the roundtripping between Password and SecurePassword... such tests will fail until the previously mentioned issues are addressed.